src/obstacle_avoidance/src/MyNode.cpp

```diff
@@ -0,0 +1,231 @@
+/*
+ * Created By: Gareth Ellis
+ * Created On: July 16th, 2016
+ * Description: An example node that subscribes to a topic publishing strings,
+ *              and re-publishes everything it receives to another topic with
+ *              a "!" at the end
+ */
+
+#include <MyNode.h>
+
+sensor_msgs::LaserScan AvoidObstacleClass::lidar_data;
+
+int sizeOfLidarArray(sensor_msgs::LaserScan lidar);
+
+AvoidObstacleClass::AvoidObstacleClass() {
+  // initialise velocity
+  vel_msg.linear.x = LINEAR_VEL;
+  vel_msg.linear.y = 0;
+  vel_msg.linear.z = 0;
+  vel_msg.angular.x = 0;
+  vel_msg.angular.y = 0;
+  vel_msg.angular.z = 0;
+}
+
+AvoidObstacleClass::AvoidObstacleClass(int argc, char **argv,
+                                       std::string node_name) {
+  // Setup NodeHandles
+  ros::init(argc, argv, node_name);
+  nh = new ros::NodeHandle;
```

Comment:
Once `nh` is not a pointer, when this constructor is called, it will automatically will be initialised with it's default constructor (so you can just remove this line). If you want to be certain of the constructor that is used to initialised `nh`, you can replace this line with `ros::NodeHandle nh;`